src/Components/articlesPage.jsx

```diff
@@ -1,13 +1,35 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import ArticleCard from "./articleCard";
+import { useLocation } from "react-router-dom";
 
 const ArticlesPage = ({ articles }) => {
+  const [filteredArticles, setFilteredArticles] = useState([]);
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const topic = searchParams.get("topic");
+
+  useEffect(() => {
+    const filteredArticlesByTopic = () => {
+      if (topic) {
+        const filtered = articles.filter((article) =>
+          article.topic && article.topic.includes(topic)
+        );
+        setFilteredArticles(filtered);
+      } else {
+        setFilteredArticles(articles);
+      }
+    };
+    filteredArticlesByTopic();
```

Comment:
This filtering could and should be done by making a new request to your API with a topic query eg `https://crowleysnewsapi.onrender.com/api/articles?topic=football`